src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/CodeAnaylsisHelper.cs

```diff
@@ -0,0 +1,37 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.MSBuild;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.MSIdentity.CodeReaderWriter
+{
+    public class CodeAnaylsisHelper
+    {
+        private string _projectFilePath;
+        public CodeAnaylsisHelper(string projectFilePath)
+        {
+            _projectFilePath = projectFilePath;
+        }
+
+        private CodeAnalysis.Project? _codeAnalysisProject;
+        public CodeAnalysis.Project CodeAnalysisProject
+        {
+            get
+            {
+                if (_codeAnalysisProject == null)
+                {
+                    _codeAnalysisProject =  LoadCodeAnalysisProject().Result;
```

Comment:
>Result [](http://example.com/codeflow?start=70&length=6)

we don't use Result in general since it can block threads - in your calling code aren't you inside the Async method? You could just have GetProjectAsync and inside the method maintain your singleton logic

Reply:
fixed and moved https://github.com/dotnet/Scaffolding/pull/1593